### PR TITLE
Add PeerConnection::GetAudioLevels

### PR DESCRIPTION
**Pull request by peter-signal**

### CODE REVIEW

File: media/engine/webrtc_voice_engine.cc [hunk 1]
```diff
@@ -2582,4 +2592,29 @@ void WebRtcVoiceMediaChannel::ConfigureEncoders(const webrtc::AudioEncoder::Conf
...
+    cricket::AudioLevel* captured_out,
+    cricket::ReceivedAudioLevel* received_out,
+    size_t received_out_size,
+    size_t* received_size_out) {
+
+  cricket::AudioLevel captured = 0;
+  for (const auto& kv : send_streams_) {
+    captured = kv.second->GetAudioLevel();
+  }
+  RTC_LOG(LS_INFO) << "FLUFF %%% captured: " << captured;
```
Comment (jim-signal) [inline media/engine/webrtc_voice_engine.cc:2605]: Do you want the log with the FLUFF or not at all?

File: media/engine/webrtc_voice_engine.cc [hunk 2]
```diff
@@ -2582,4 +2592,29 @@ void WebRtcVoiceMediaChannel::ConfigureEncoders(const webrtc::AudioEncoder::Conf
...
+
+  size_t received_size = 0;
+  for (const auto& kv : recv_streams_) {
+    received_out[received_size++] = cricket::ReceivedAudioLevel {
```
Comment (jim-signal) [inline media/engine/webrtc_voice_engine.cc:2609]: Should received_out_size to be used for defense/validity?

File: media/engine/webrtc_voice_engine.cc [hunk 3]
```diff
@@ -2582,4 +2592,29 @@ void WebRtcVoiceMediaChannel::ConfigureEncoders(const webrtc::AudioEncoder::Conf
...
 
+void WebRtcVoiceMediaChannel::GetAudioLevels(
+    cricket::AudioLevel* captured_out,
+    cricket::ReceivedAudioLevel* received_out,
+    size_t received_out_size,
+    size_t* received_size_out) {
+
+  cricket::AudioLevel captured = 0;
+  for (const auto& kv : send_streams_) {
+    captured = kv.second->GetAudioLevel();
```
Comment (jim-signal) [inline media/engine/webrtc_voice_engine.cc:2603]: The expectation is that there will only really be one send stream?

File: media/engine/webrtc_voice_engine.cc (hunk 1 above)
Comment (peter-signal) [reply media/engine/webrtc_voice_engine.cc:2605]: Removed

File: media/engine/webrtc_voice_engine.cc (hunk 2 above)
Comment (peter-signal) [reply media/engine/webrtc_voice_engine.cc:2611]: Yes, done

File: media/engine/webrtc_voice_engine.cc (hunk 3 above)
Comment (peter-signal) [reply media/engine/webrtc_voice_engine.cc:2603]: Yes.

**Review by jim-signal — APPROVED** [review]

**Pull request merged**